File: Harvest/Models/Seed.swift
import Foundation

enum SeedStatus: String, Codable {
    case pending, accepted, declined
}

struct Seed: Identifiable, Codable, Equatable {
    let id: String
    let senderId: String
    let recipientId: String
    let openingMessage: String
    let status: SeedStatus
    let conversationId: String?
    let createdAt: String?
    let respondedAt: String?

    enum CodingKeys: String, CodingKey {
        case id
        case senderId = "sender_id"
        case recipientId = "recipient_id"
        case openingMessage = "opening_message"
        case status
        case conversationId = "conversation_id"
        case createdAt = "created_at"
        case respondedAt = "responded_at"
    }
}

File: android/app/src/main/java/com/harvestglass/harvest/data/model/Seed.kt
package com.harvestglass.harvest.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Mirrors Harvest/Models/Seed.swift. */
@Serializable
enum class SeedStatus {
    @SerialName("pending") PENDING,
    @SerialName("accepted") ACCEPTED,
    @SerialName("declined") DECLINED
}

@Serializable
data class Seed(
    val id: String,
    @SerialName("sender_id") val senderId: String,
    @SerialName("recipient_id") val recipientId: String,
    @SerialName("opening_message") val openingMessage: String,
    val status: SeedStatus,
    @SerialName("conversation_id") val conversationId: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
    @SerialName("responded_at") val respondedAt: String? = null
)
